SF iOS/SF iOS/Models/EventDataSource.m

```diff
@@ -95,6 +103,10 @@ - (void)refresh {
 
         [realm transactionWithBlock:^{
             [realm addOrUpdateObjects:addToRealm];
+            dispatch_async(dispatch_get_main_queue(), ^{
+                // Reload UI on the main queue
+                [self.delegate reloadData];
```

Comment:
do we need a weak reference here? not sure around the timing of Realm updates or what triggers them